lib/money/allocator.rb

```diff
@@ -51,11 +59,26 @@ def allocate(splits, strategy = :roundrobin)
 
       amounts, left_over = amounts_from_splits(allocations, splits)
 
+      order = case strategy
+      when :roundrobin
+        # returns [0,1,2 etc]
+        (0...left_over).to_a
+      when :roundrobin_reverse
+        first = amounts.length - left_over
+        last = amounts.length
+        # returns [5,4,3 etc]
+        (first...last).to_a.reverse
```

Comment:
Done. I had initially ruled out such a solution, because it results in the `order` variable being larger than the number of leftover pennies, e.g. you could have `order = [7, 6, 5, 4, 3, 2, 1]` and only 1 leftover penny. But thinking about it more now, I realize this does not actually matter because, the number of iterations is constrained on [line 73](https://github.com/Shopify/money/blob/469438a221118cef3be366c6845c2ac8df83bfcf/lib/money/allocator.rb#L73-L75).